### PR TITLE
Upgrade SDK for improved Accounts service

### PR DESCRIPTION
**Pull request by zyro**
* Upgrade to Android Gradle plugin 1.0.1.
* Upgrade to Android appcompat-v7 21.0.3.
* Upgrade to OkHttp 2.2.0.
* Greatly improve SDK interface and structure.
* Update all calls to reflect new Accounts service.
* Update `Gamer` entity to reflect single 'name' field from `/v0/gamer`.
* API keys are now set once.
* `GameUpSession` instances track their own API keys.
* Better `WebView` error handling.
* Basic retries for REST calls to both Game and Accounts endpoints.
* UDID is not being handled implicitly. Instead, the `GameUp.loginAnonymous()` method expects to be given the ID to use.

### CODE REVIEW

**Comment by mofirouz** [comment]
Good stuff @zyro. Big :+1:

File: sdk/src/main/java/io/gameup/android/GameUpException.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+/**
+ * Details of an error response from the server, or a wrapped client-side error.
+ */
+@Data
+@EqualsAndHashCode(callSuper = false)
+@AllArgsConstructor(suppressConstructorProperties = true)
+public class GameUpException extends RuntimeException {
+
+    private final int status;
```
Comment (novabyte) [inline sdk/src/main/java/io/gameup/android/GameUpException.java]: Doccomment please.

File: sdk/src/main/java/io/gameup/android/GameUpException.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    private final String message;
```
Comment (novabyte) [inline sdk/src/main/java/io/gameup/android/GameUpException.java]: Here too.

File: sdk/src/main/java/io/gameup/android/GameUpException.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    private final RequestInfo request;
```
Comment (novabyte) [inline sdk/src/main/java/io/gameup/android/GameUpException.java]: Here too.

File: sdk/src/main/java/io/gameup/android/json/LoginAnonymousRequest.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import lombok.Data;
+
+/**
+ * Represents a request to the anonymous login endpoint.
+ */
+@Data
+@AllArgsConstructor(suppressConstructorProperties = true)
+public class LoginAnonymousRequest {
+
+    private final String id;
```
Comment (novabyte) [inline sdk/src/main/java/io/gameup/android/json/LoginAnonymousRequest.java]: Doccomment please.

File: sdk/src/main/java/io/gameup/android/json/LoginOAuthRequest.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import lombok.Data;
+
+/**
+ * Represents a request to the OAuth passthrough login endpoint.
+ */
+@Data
+@AllArgsConstructor(suppressConstructorProperties = true)
+public class LoginOAuthRequest {
+
+    private final String type;
```
Comment (novabyte) [inline sdk/src/main/java/io/gameup/android/json/LoginOAuthRequest.java]: Doccomment please.

File: sdk/src/main/java/io/gameup/android/json/LoginOAuthRequest.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+    private final String accessToken;
```
Comment (novabyte) [inline sdk/src/main/java/io/gameup/android/json/LoginOAuthRequest.java]: Here too.

File: sdk/src/main/java/io/gameup/android/json/LoginResponse.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+import lombok.Data;
+
+/**
+ * Represents a successful response from a REST login endpoint.
+ */
+@Data
+@AllArgsConstructor(suppressConstructorProperties = true)
+public class LoginResponse {
+
+    private final String token;
```
Comment (novabyte) [inline sdk/src/main/java/io/gameup/android/json/LoginResponse.java]: Doccomment please.

**Pull request closed without merging**